### ISSUE TITLE
Create a Job Summary API

### ISSUE THREAD

**Issue by dadgar** [hashicorp/nomad#1340]
Have a Job Summary API that will list the number of allocations in the following state:

* Pending
* Starting
* Running
* Complete
* Failed

**Comment by a86c6f7964** [comment]
My ideal api (in keeping as similar look and feel as the existing api) would look something like
```
$ nomad status
<every job>
==> Job Summary
Pending   = 10
Starting  = 0
...

$ nomad status some-prefix
<every job matching that prefix>
==> Job Summary
...

$ nomad status some-exact-id
<current nomad status output>
==> Job Summary
...
````

also, it would be nice to add a param such as `-machine` for an output that is a little easier to parse with awk and grep and such.

**Comment by dadgar** [comment]
What does the first output represent? The rolled up state of the whole cluster?

**Comment by a86c6f7964** [comment]
Yes

**Comment by dadgar** [comment]
Fixed by #1455
